app/src/main/java/cm/aptoide/pt/ApplicationModule.java

```diff
@@ -867,6 +870,18 @@ public ApplicationModule(AptoideApplication application, String aptoideMd5sum) {
     return RxJavaCallAdapterFactory.create();
   }
 
+  @Singleton @Provides SearchManager providesSearchManager(@Named("pool-v7")
+      BodyInterceptor<cm.aptoide.pt.dataprovider.ws.v7.BaseBody> baseBodyBodyInterceptor,
+      @Named("default") SharedPreferences sharedPreferences, TokenInvalidator tokenInvalidator,
+      @Named("default") OkHttpClient okHttpClient, Converter.Factory converterFactory,
+      AdsRepository adsRepository) {
+    return new SearchManager(sharedPreferences, tokenInvalidator, baseBodyBodyInterceptor,
+        okHttpClient, converterFactory, StoreUtils.getSubscribedStoresAuthMap(
+        AccessorFactory.getAccessorFor(application.getDatabase(), Store.class)),
+        StoreUtils.getSubscribedStoresIds(
+            AccessorFactory.getAccessorFor(application.getDatabase(), Store.class)), adsRepository);
```

Comment:
Database has a provider already so you can you use it, instead of doing application.getDatabase();

Reply:
Fixed